vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/occupancy_handler.rb

```diff
@@ -47,6 +47,13 @@ def new_occupancy
         )
       end
 
+      def complete_existing_occupancy!
+        existing_occupancy.mark_orphaned!
+        if existing_occupancy.order_completable?
+          existing_occupancy.order_detail.complete!
```

Comment:
Also, is there a way to move this order logic into the order handler? (Well, it looks like you have similar logic already. Can we dry it up a little?)